program-runtime/src/invoke_context.rs

```diff
@@ -182,6 +231,10 @@ impl<'a> InvokeContext<'a> {
             current_compute_budget: compute_budget,
             compute_budget,
             compute_meter: ComputeMeter::new_ref(compute_budget.max_units),
+            accounts_data_meter: AccountsDataMeter::new_ref(
+                // SAFETY: The assert above guarantees the subtraction operation will not wrap
```

Comment:
nit: it would be better to locate this comment/line immediately following the assert to defend against future code refactoring.  One example:
```rust
let accounts_data_meter = {
    assert!(current_accounts_data_len <= MAX_ACCOUNTS_DATA_LEN);
    AccountsDataMeter::new_ref(
        // SAFETY: The assert above guarantees the subtraction operation will not wrap
        MAX_ACCOUNTS_DATA_LEN.wrapping_sub(current_accounts_data_len),
    )
};
```
then just include `accounts_data_meter` in the `Self` initializer

Reply:
!!! This is excellent. I was not happy that my assert was so far away. Thanks for the help!